### PR TITLE
Vcpkg cmake presets

### PR DESCRIPTION
**Pull request by ekoby**
- use vcpkg to manage deps
- use CMake presets to manage build configurations
- update ziti-sdk to v0.32.0

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -98,22 +98,23 @@ jobs:
...
+      - name: install contemporary cmake
+        uses: lukka/get-cmake@latest
 
-      - name: configure cmake
-        env:
-          TOOLCHAIN: ${{ matrix.toolchain && format('./toolchains/{0}', matrix.toolchain) || '' }}
-        run: cmake ${{matrix.cmake_opts}} -DCMAKE_TOOLCHAIN_FILE="${TOOLCHAIN}" -S . -B ./build
+      - uses: lukka/run-vcpkg@v10
+        with:
+          vcpkgGitCommitId: '19af97cba8ca48474e4ad15a24ed50271a9ecdac'
```
Comment (qrkourier) [inline .github/workflows/cmake.yml:111]: @ekoby Will you please add a note about this commit ID so we'll understand its significance and to inform future changes to the value?

**Review by qrkourier — CHANGES_REQUESTED** [review]

**Comment by qrkourier** [comment]
@ekoby I have started a package build on this branch to test.

https://github.com/openziti/ziti-tunnel-sdk-c/actions/workflows/cpack.yml?query=branch%3Avcpkg-cmake-presets

**Comment by ekoby** [comment]
> @ekoby I have started a package build on this branch to test.
> 
> https://github.com/openziti/ziti-tunnel-sdk-c/actions/workflows/cpack.yml?query=branch%3Avcpkg-cmake-presets

@qrkourier this needs to be changed to add run-vcpkg step and use an appropriate CMake preset.

We can probably refactor those steps into one action (similar to ziti-sdk-c)

**Comment by scareything** [comment]
the windows-arm64 archive contains amd64 stuff.
image

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -129,7 +132,7 @@ jobs:
       - name: upload bundle artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: ${{ runner.os }}-${{ matrix.arch }}
+          name: ${{ matrix.target }}
           path: |
             ./build/bundle/ziti-edge-tunnel-*.zip
```
Comment (scareything) [inline .github/workflows/cmake.yml:137]: We can exclude the test binaries with a negation:

https://github.com/openziti/ziti-tunnel-sdk-c/pull/623/files#diff-cdd48abbd3eb8d1c54077449fc74a8de1f29805d2be5d8e5232b7aab76ea7a6fR148

Comment (ekoby) [reply .github/workflows/cmake.yml:137]: better yet, do not bundle ziti-edge-tunnel-test at all

**Review by scareything — APPROVED** [review]
looks ok to me. I do wish it were possible to build without creating or using a preset.

**Pull request merged**